admin/spam/views.py

```diff
@@ -1,96 +1,124 @@
 from __future__ import unicode_literals
 
 from django.shortcuts import render
-from django.contrib.auth.decorators import login_required
 from django.core.mail import send_mail
-from django.http import HttpResponseNotFound
 from django.views.generic import FormView, ListView
-from django.utils.decorators import method_decorator
-from django.core.urlresolvers import reverse
 from django.views.defaults import page_not_found
 
 from modularodm import Q
 from website.project.model import Comment
 from website.settings import SUPPORT_EMAIL
 
-from admin.common_auth.logs import update_admin_log, CONFIRM_HAM, CONFIRM_SPAM
+from admin.base.utils import OSFAdmin
+from admin.common_auth.logs import (
+    update_admin_log,
+    CONFIRM_HAM,
+    CONFIRM_SPAM,
+    USER_EMAILED,
+)
 from admin.spam.serializers import serialize_comment
 from admin.spam.forms import EmailForm, ConfirmForm
+from admin.spam.templatetags.spam_extras import reverse_spam_detail
```

Comment:
I saw these template tags. It seems like you could consolidate them into one tag that takes a view name, kwargs, and querystring kwargs. Sorta like this but one that supports more than one qs kwarg: 
```python

from django.utils.http import urlencode

def reverse_querystring(view, urlconf=None, args=None, kwargs=None, current_app=None, query_kwargs=None):
    '''Custom reverse to handle query strings.
    Usage:
        reverse('app.views.my_view', kwargs={'pk': 123}, query_kwargs={'search', 'Bob'})
    '''
    base_url = reverse(view, urlconf=urlconf, args=args, kwargs=kwargs, current_app=current_app)
    if query_kwargs:
        return '{}?{}'.format(base_url, urlencode(query_kwargs)
    return base_url
```
https://gist.github.com/benbacardi/227f924ec1d9bedd242b